### PR TITLE
feat(nodejs dev deps) filter nodejs dev deps when using npm ls

### PR DESCRIPTION
**Pull request by microsoftly**
Theres a lot of test/fixture work here that effectively amounts to being unnecessary for this fix, but I created it proactively to prove correctness.

The problem is unit tests did not pick up the proper state because they mocked out `npm ls` and shared a single file that was the output of `npm ls`. To make each fixture more unique, I saved the output of `npm ls` to each nodejs/testdata/chai fixture where it would be appropriate. 

I had run it with `npm ls --json --production`, which, as it turns out resulted in the same output for each case, including dev deps. This led me to the solution that found the bug: we were leaving out `--production` in our call.

That being said, the test refactors are still functionally useful as they keep each fixture separate, and also add cases against which tests are run (e.g. multiple fixtures per test type). For that reason, I have kept them in this PR.

Also captured is the cleanup of unneeded files in the fixtures (e.g. `LICENSE` and `CODEOWNERS`

### CODE REVIEW

File: analyzers/nodejs/testdata/chai/dev-deps/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "testdata",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "MIT",
+  "dependencies": {
+    "chai": "4.1.2"
+  },
+  "devDependencies": {
+    "is-odd": "^3.0.1"
```
Comment (liftM) [inline analyzers/nodejs/testdata/chai/dev-deps/package.json]: See my same comments on the Yarn parsing re: fixtures.

File: analyzers/nodejs/nodejs_test.go [hunk 2]
```diff
@@ -94,8 +94,11 @@ var chaiDirectDep = pkg.Import{
 }
 
 func TestAnalyzeWithNpmLs(t *testing.T) {
-	buildTarget := filepath.Join("testdata", "chai", "installed")
-
+	testAnalyzeWithNpmLs(t, filepath.Join("testdata", "chai", "installed"))
```
Comment (liftM) [inline analyzers/nodejs/nodejs_test.go]: Let's just make this a table-driven `for` loop.

File: analyzers/nodejs/nodejs_test.go [hunk 3]
```diff
@@ -118,8 +121,12 @@ func TestAnalyzeWithNpmLs(t *testing.T) {
 }
 
 func TestUsingNodeModuleFallback(t *testing.T) {
-	buildTarget := filepath.Join("testdata", "chai", "installed")
+	testUsingNodeModuleFallback(t, filepath.Join("testdata", "chai", "installed"))
```
Comment (liftM) [inline analyzers/nodejs/nodejs_test.go]: Same here as above.

**Comment by microsoftly** [comment]
Will provide new fixture strategy in a follow up PR

**Pull request merged**